src/mlir/generator.cc

```diff
@@ -68,17 +68,14 @@ namespace mlir::verona
       return runtimeError(
         "Cannot open file " + filename + ": " + err.message());
 
-    // Setup source manager and parse
+    // Setup source manager and parse the input. This includes verification of
+    // the IR.
     llvm::SourceMgr sourceMgr;
     sourceMgr.AddNewSourceBuffer(std::move(*srcOrErr), llvm::SMLoc());
     module = mlir::parseSourceFile(sourceMgr, builder.getContext());
+    if (!module)
```

Comment:
Dumping the IR on failure is quite helpful, especially at this stage.

Also, no reason to change the error message.

Reply:
We don't have access to the IR is any of the verifiers fail, `parseSourceFile` return null. There's nothing we could dump.

I've changed the error message intenitonally because this case happens on both parsing and verification errors, and there's not much we can do to distinguish the two. "Load" felt like a generalization of the two steps.

Reply:
Right, this is reading MLIR...